Rust CI cache causes build failures every few months, fixed by deleting cache

See https://github.com/ddnet/ddnet/actions/runs/22457710020/job/65043264161?pr=11845

<details>

```
    Updating crates.io index
   Compiling mastersrv v0.0.1 (/Users/runner/work/ddnet/ddnet/src/mastersrv)
error[E0460]: found possibly newer version of crate `std` which `log` depends on
  --> src/main.rs:44:1
   |
44 | extern crate log;
   | ^^^^^^^^^^^^^^^^^
   |
   = note: perhaps that crate needs to be recompiled?
   = note: the following crate versions were found:
           crate `std`: /opt/homebrew/Cellar/rust/1.93.1_1/lib/rustlib/aarch64-apple-darwin/lib/libstd-07c7968069e7a5c2.rlib
           crate `std`: /opt/homebrew/Cellar/rust/1.93.1_1/lib/rustlib/aarch64-apple-darwin/lib/libstd-07c7968069e7a5c2.dylib
           crate `log`: /Users/runner/work/ddnet/ddnet/src/mastersrv/target/debug/deps/liblog-9c221149f96df500.rlib

error[E0463]: can't find crate for `arc_swap`
 --> src/main.rs:1:5
  |
1 | use arc_swap::ArcSwap;
  |     ^^^^^^^^ can't find crate

error[E0463]: can't find crate for `arrayvec`
 --> src/main.rs:2:5
  |
2 | use arrayvec::ArrayString;
  |     ^^^^^^^^ can't find crate

error[E0463]: can't find crate for `clap`
 --> src/main.rs:4:5
  |
4 | use clap::value_t_or_exit;
  |     ^^^^ can't find crate

error[E0463]: can't find crate for `headers`
 --> src/main.rs:7:5
  |
7 | use headers::HeaderMapExt as _;
  |     ^^^^^^^ can't find crate

error[E0463]: can't find crate for `rand`
 --> src/main.rs:8:5
  |
8 | use rand::random;
  |     ^^^^ can't find crate

error[E0463]: can't find crate for `serde`
 --> src/main.rs:9:5
  |
9 | use serde::Deserialize;
  |     ^^^^^ can't find crate

error[E0463]: can't find crate for `serde_json`
  --> src/main.rs:11:5
   |
11 | use serde_json as json;
   |     ^^^^^^^^^^ can't find crate

error[E0463]: can't find crate for `sha2`
  --> src/main.rs:12:5
   |
12 | use sha2::Digest;
   |     ^^^^ can't find crate

error[E0463]: can't find crate for `tokio`
  --> src/main.rs:36:5
   |
36 | use tokio::fs;
   |     ^^^^^ can't find crate

error[E0463]: can't find crate for `warp`
  --> src/main.rs:40:5
   |
40 | use warp::Filter;
   |     ^^^^ can't find crate

error[E0463]: can't find crate for `url`
 --> src/addr.rs:7:5
  |
7 | use url::Url;
  |     ^^^ can't find crate

error[E0463]: can't find crate for `constant_time_eq`
 --> src/community_token.rs:2:5
  |
2 | use constant_time_eq::constant_time_eq;
  |     ^^^^^^^^^^^^^^^^ can't find crate

error[E0463]: can't find crate for `ipnet`
 --> src/config.rs:5:5
  |
5 | use ipnet::IpNet;
  |     ^^^^^ can't find crate

error[E0463]: can't find crate for `tokio_stream`
    --> src/main.rs:1231:17
     |
1231 |             use tokio_stream::wrappers::UnixListenerStream;
     |                 ^^^^^^^^^^^^ can't find crate

error: cannot find attribute `serde` in this scope
  --> src/config.rs:39:3
   |
39 | #[serde(deny_unknown_fields)]
   |   ^^^^^

error: cannot find attribute `serde` in this scope
  --> src/config.rs:41:7
   |
41 |     #[serde(default)]
   |       ^^^^^

error: cannot find attribute `serde` in this scope
  --> src/config.rs:43:7
   |
43 |     #[serde(default)]
   |       ^^^^^

error: cannot find attribute `serde` in this scope
  --> src/config.rs:45:7
   |
45 |     #[serde(default)]
   |       ^^^^^

error: cannot find attribute `serde` in this scope
  --> src/config.rs:47:7
   |
47 |     #[serde(default)]
   |       ^^^^^

error: cannot find attribute `serde` in this scope
  --> src/config.rs:49:7
   |
49 |     #[serde(default)]
   |       ^^^^^

error: cannot find attribute `serde` in this scope
  --> src/config.rs:56:7
   |
56 |     #[serde(default)]
   |       ^^^^^

error: cannot find attribute `serde` in this scope
  --> src/config.rs:61:3
   |
61 | #[serde(deny_unknown_fields)]
   |   ^^^^^

error: cannot find attribute `serde` in this scope
  --> src/main.rs:84:3
   |
84 | #[serde(rename_all = "snake_case", tag = "status")]
   |   ^^^^^

error: cannot find attribute `serde` in this scope
  --> src/main.rs:94:7
   |
94 |     #[serde(skip)]
   |       ^^^^^

error: cannot find attribute `serde` in this scope
   --> src/main.rs:177:7
    |
177 |     #[serde(skip_serializing_if = "Option::is_none")]
    |       ^^^^^

error: cannot find attribute `serde` in this scope
   --> src/main.rs:194:7
    |
194 |     #[serde(skip_serializing_if = "Option::is_none")]
    |       ^^^^^

error: cannot find attribute `serde` in this scope
   --> src/main.rs:196:7
    |
196 |     #[serde(skip_serializing_if = "Option::is_none")]
    |       ^^^^^

error: cannot find attribute `serde` in this scope
   --> src/main.rs:216:7
    |
216 |     #[serde(skip_serializing_if = "Option::is_none")]
    |       ^^^^^

error: cannot find attribute `serde` in this scope
   --> src/main.rs:265:3
    |
265 | #[serde(rename_all = "snake_case")]
    |   ^^^^^

error: cannot find attribute `serde` in this scope
   --> src/main.rs:275:7
    |
275 |     #[serde(skip_serializing_if = "Option::is_none")]
    |       ^^^^^

error: cannot find macro `debug` in this scope
   --> src/main.rs:474:17
    |
474 |                 debug!("removing {} due to timeout", addr);
    |                 ^^^^^

error: cannot find macro `error` in this scope
   --> src/main.rs:730:21
    |
730 |                     error!("error re-reading config: {}", e);
    |                     ^^^^^

error: cannot find macro `info` in this scope
   --> src/main.rs:735:21
    |
735 |                     info!("successfully reloaded config");
    |                     ^^^^

error: cannot find macro `debug` in this scope
   --> src/main.rs:839:33
    |
839 |             AddResult::Added => debug!("successfully registered {}", addr),
    |                                 ^^^^^

error: cannot find macro `warn` in this scope
   --> src/main.rs:843:17
    |
843 |                 warn!(
    |                 ^^^^
    |
    = note: `warn` is in scope, but it is an attribute: `#[warn]`

error: cannot find macro `trace` in this scope
   --> src/main.rs:860:13
    |
860 |             trace!("re-sending challenge to {}", addr);
    |             ^^^^^

error: cannot find macro `trace` in this scope
   --> src/main.rs:862:13
    |
862 |             trace!("sending challenge to {}", addr);
    |             ^^^^^

error: cannot find macro `debug` in this scope
   --> src/main.rs:884:13
    |
884 |             debug!("successfully removed {}", addr);
    |             ^^^^^

error: cannot find macro `info` in this scope
    --> src/main.rs:1102:21
     |
1102 |                     info!("successfully read previous dump");
     |                     ^^^^

error: cannot find macro `error` in this scope
    --> src/main.rs:1105:39
     |
1105 |                 Err(FromDumpError) => error!("previous dump was inconsistent"),
     |                                       ^^^^^

error: cannot find macro `error` in this scope
    --> src/main.rs:1107:23
     |
1107 |             Err(e) => error!("couldn't read previous dump: {}", e),
     |                       ^^^^^

error[E0463]: can't find crate for `base64`
  --> src/community_token.rs:14:9
   |
14 |         base64::decode_config_slice(encoded_crc32, base64::URL_SAFE_NO_PAD, &mut crc32).ok()?,
   |         ^^^^^^ can't find crate

error[E0463]: can't find crate for `crc32fast`
  --> src/community_token.rs:17:8
   |
17 |     if crc32fast::hash(result.as_bytes()).to_be_bytes() != crc32 {
   |        ^^^^^^^^^ can't find crate

error[E0463]: can't find crate for `toml`
   --> src/config.rs:191:9
    |
191 |         toml::from_str(&contents).map_err(|e| Error(format!("TOML error in {:?}: {}", filename, e)))
    |         ^^^^ can't find crate

error[E0463]: can't find crate for `libloc`
  --> src/locations.rs:14:19
   |
14 |     inner: Option<libloc::Locations>,
   |                   ^^^^^^ can't find crate

error[E0463]: can't find crate for `hex`
   --> src/main.rs:778:13
    |
778 |             hex::decode_to_slice(token_hex.as_bytes(), &mut token).map_err(|e| {
    |             ^^^ can't find crate

error[E0463]: can't find crate for `mime`
   --> src/main.rs:931:22
    |
931 |                 .map(mime::Mime::from)
    |                      ^^^^ can't find crate

error[E0463]: can't find crate for `env_logger`
   --> src/main.rs:998:5
    |
998 |     env_logger::init();
    |     ^^^^^^^^^^ can't find crate

error[E0463]: can't find crate for `bytes`
    --> src/main.rs:1179:83
     |
1179 |             move |headers: warp::http::HeaderMap, addr: Option<SocketAddr>, info: bytes::Bytes| {
     |                                                                                   ^^^^^ can't find crate

error[E0220]: associated type `Ok` not found for `S`
  --> src/addr.rs:62:56
   |
62 |     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
   |                                                        ^^ associated type `Ok` not found

error[E0220]: associated type `Error` not found for `S`
  --> src/addr.rs:62:63
   |
62 |     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
   |                                                               ^^^^^ associated type `Error` not found

error[E0220]: associated type `Error` not found for `D`
  --> src/addr.rs:85:63
   |
85 |     fn deserialize<D>(deserializer: D) -> Result<Protocol, D::Error>
   |                                                               ^^^^^ associated type `Error` not found

error[E0220]: associated type `Ok` not found for `S`
   --> src/addr.rs:147:56
    |
147 |     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
    |                                                        ^^ associated type `Ok` not found

error[E0220]: associated type `Error` not found for `S`
   --> src/addr.rs:147:63
    |
147 |     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
    |                                                               ^^^^^ associated type `Error` not found

error[E0220]: associated type `Error` not found for `D`
   --> src/addr.rs:172:59
    |
172 |     fn deserialize<D>(deserializer: D) -> Result<Addr, D::Error>
    |                                                           ^^^^^ associated type `Error` not found

error[E0220]: associated type `Ok` not found for `S`
   --> src/addr.rs:246:56
    |
246 |     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
    |                                                        ^^ associated type `Ok` not found

error[E0220]: associated type `Error` not found for `S`
   --> src/addr.rs:246:63
    |
246 |     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
    |                                                               ^^^^^ associated type `Error` not found

error[E0220]: associated type `Error` not found for `D`
   --> src/addr.rs:271:67
    |
271 |     fn deserialize<D>(deserializer: D) -> Result<RegisterAddr, D::Error>
    |                                                                   ^^^^^ associated type `Error` not found

error[E0220]: associated type `Error` not found for `D`
   --> src/config.rs:295:32
    |
295 |     ) -> Result<ConfigAddr, D::Error> {
    |                                ^^^^^ associated type `Error` not found

Some errors have detailed explanations: E0220, E0460, E0463.
For more information about an error, try `rustc --explain E0220`.
error: could not compile `mastersrv` (bin "mastersrv") due to 60 previous errors
Error: Process completed with exit code 101.
```

</details>

I think this happened a few times before already. I fixed it by deleting the Rust cache entry that was being used by this job (see https://github.com/ddnet/ddnet/actions/runs/22457710020/job/65043264161?pr=11845#step:9:63) in https://github.com/ddnet/ddnet/actions/caches.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -103,7 +103,7 @@ jobs:
       if: contains(matrix.os, 'macOS')
       run: |
         brew update
-        brew install ffmpeg molten-vk ninja pkg-config rust sdl2
+        brew install ffmpeg molten-vk ninja pkg-config sdl2
         brew upgrade freetype
         pip3 install --break-system-packages dmgbuild
         echo /Library/Frameworks/Python.framework/Versions/3.12/bin >> $GITHUB_PATH

